helsingborg-stad/helsingborg-io-sls-api
# Fix/handle undefined address

## Explain the changes you’ve made
Handle navet users which does not have an address registered in Navet

## Explain why these changes are made
If a user didn't had an address registered in Navet and tried to login to the application, the user would be signed out immediately. That wasn't the wanted behavior, so instead on throwing in the `pollUser` lambda, we now handle addresses which doesn't exists.

## Explain your solution
The lambda now defaults the addresses properties to undefined if they do not exists

## How to test

1. Checkout this branch
2. run `sls deploy` within service folder for `service navet-ms`
3. Make sure you have a user registered in Navet that you can test with
4. Remove user if exists in the DynamoDB table
5. Login to the application, the user should appear in the DynamoDB table
6. Check the profile page in the application and make sure everything except address are filled

## Tested environments

- [] Your local Serverless environment.
